VS nuget restore is not respecting NoWarn on package references

When I restore in VS, NoWarn on PackageReference has no effect.

When I restore via msbuild /t:restore, it works as expected. 

This is a regression in the 15.6 dogfood build I have. It works fine on VS 15.5.1 RTM.

## Details about Problem

* **NuGet product used (NuGet.exe | VS UI | Package Manager Console | dotnet.exe)**: VS UI
* **NuGet version (x.x.x.xxx)**: 4.6.0.??? (VS About doesn't show 4th part)
* **VS version (if appropriate):** 15.6.0 Preview 1.1 [27217.0.d15.6]
* **Worked before? If so, with which NuGet version**: Yes, worked fine in VS 15.5.1 RTM.


## Detailed repro steps so we can see the same problem

1. Build this project with restore having been done in VS:

``` xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="NewtonSoft.Json" Version="4.5.11" NoWarn="NU1701" />
  </ItemGroup>
</Project>
```

## Expected result

Builds without warnings

## Actual result
```
1>c:\users\nicholg\source\repos\ConsoleApp22\ConsoleApp22\ConsoleApp22.csproj : warning NU1701: Package 'Newtonsoft.Json 4.5.11' was 
restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be 
fully compatible with your project.
```

cc @emgarten @davkean

--- a/src/NuGet.Core/NuGet.LibraryModel/LibraryDependency.cs
+++ b/src/NuGet.Core/NuGet.LibraryModel/LibraryDependency.cs
@@ -1,4 +1,4 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
@@ -99,7 +99,8 @@ namespace NuGet.LibraryModel
                 },
                 SuppressParent = SuppressParent,
                 Type = Type,
-                AutoReferenced = AutoReferenced
+                AutoReferenced = AutoReferenced,
+                NoWarn = NoWarn
             };
         }
     }

--- a/test/NuGet.Core.Tests/NuGet.ProjectModel.Test/PackageSpecCloningTests.cs
+++ b/test/NuGet.Core.Tests/NuGet.ProjectModel.Test/PackageSpecCloningTests.cs
@@ -594,6 +594,7 @@ namespace NuGet.ProjectModel.Test
             dependency.Type = LibraryDependencyType.Default;
             dependency.IncludeType = LibraryIncludeFlags.None;
             dependency.SuppressParent = LibraryIncludeFlags.ContentFiles;
+            dependency.NoWarn = new List<NuGetLogCode>() { NuGetLogCode.NU1000, NuGetLogCode.NU1001 };
             var imports = NuGetFramework.Parse("net45"); // This makes no sense in the context of fallback, just for testing :)
 
             var originalTargetFrameworkInformation = new TargetFrameworkInformation();

